packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -152,23 +169,45 @@ class DoubleTapGestureRecognizer extends GestureRecognizer {
   ///  * [kPrimaryButton], the button this callback responds to.
   GestureDoubleTapCallback onDoubleTap;
 
+  /// Called when a pointer that will trigger a tap of a primary button has
+  /// stopped contacting the screen at a particular location, unless a previous
+  /// tap landed in proximity shortly before, in which case [onDoubleTap] and
+  /// [onDoubleTapUp] will be called.
+  ///
+  /// [onFirstTapUp] is called optimistically before winning the gesture arena,
+  /// the same pointer that triggered [onFirstTapUp] might be rejected afterwards.
+  /// Neither [onDoubleTap] or [onDoubleTapUp] is guaranteed to be called after
+  /// [onFirstTapUp].
+  GestureTapUpCallback onFirstTapUp;
+
+  /// Called when the user has tapped the screen with a primary button at the
+  /// same location twice in quick succession.
+  ///
+  /// This triggers when the pointer stops contacting the device after the
+  /// second tap, immediately after winning the gesture arenas of both taps.
+  GestureTapUpCallback onDoubleTapUp;
```

Comment:
Is this sufficient for the double tap drag use case? Don't you still need to know when and where the second tap came down?